### PR TITLE
Add "try all" button to the failing screen

### PR DESCRIPTION
**Pull request by bemasc**
This should help users recover if they can no
longer access the current selected server.

### CODE REVIEW

**Comment by bemasc** [comment]
(CI is failing due to lack of translation for new strings.  Please review; then we'll get translations and confirm CI passes before merging.)

File: Android/app/src/main/java/app/intra/ui/settings/ServerApprovalDialogFragment.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+    final int index = getIndex();
+    final String name = getResources().getStringArray(R.array.names)[index];
+    final String template = getResources().getString(R.string.found_fastest);
+    final String message = String.format(template, name);
+
+    final String link = getResources().getStringArray(R.array.server_websites)[index];
+    final boolean showWebsite = getShowWebsite();
+    final View body;
+    if (showWebsite) {
```
Comment (alalamav) [inline Android/app/src/main/java/app/intra/ui/settings/ServerApprovalDialogFragment.java]: Consider decomposing this method to be more readable, particularly the webview and layout creation could be abstracted away.

File: Android/app/src/main/java/app/intra/net/doh/Probe.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      0, 1,  // QTYPE = A
+      0, 1   // QCLASS = IN (Internet)
+  };
+
+  private static final DnsUdpQuery QUERY = DnsUdpQuery.fromUdpBody(QUERY_DATA);
+
+  enum Status { NEW, RUNNING, SUCCEEDED, FAILED }
+  private Status status = Status.NEW;
+
+  interface Callback {
```
[the remark this answers is not in the document]
Comment (bemasc) [reply Android/app/src/main/java/app/intra/net/doh/Probe.java]: Unfortunately we don't have `CompletableFuture` on older Android versions, so we can't really use it here.  Do you see a good way to use the base `Future` interface?

Part of the problem here is that `Probe` does _both_ a slow synchronous operation (`factory.get()`) _and_ a slow async operation (`conn.performDnsRequest`).  That's due to the blocking nature of `InetAddress.getByName()` and the callback style of OkHttp3.

We could pull in a `CompletableFuture` polyfill from Guava if you think it's worth it...

File: Android/app/src/main/java/app/intra/ui/settings/ServerApprovalDialogFragment.java (hunk 1 above)
Comment (bemasc) [reply Android/app/src/main/java/app/intra/ui/settings/ServerApprovalDialogFragment.java]: Done

File: Android/app/src/main/java/app/intra/net/doh/Probe.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (alalamav) [reply Android/app/src/main/java/app/intra/net/doh/Probe.java]: Ah! Forgot `CompletableFuture`is relatively new. I think using `Future` with OkHttp3 callback style will complicate the implementation. I don't think it's worth polyfilling.

**Pull request merged**